x/authenticator/authenticator/signature_authenticator_test.go

```diff
@@ -15,101 +15,399 @@ import (
-	txConfig := encodingConfig.TxConfig
+func (s *SigVerifyAuthenticationSuite) SetupTest() {
+	// Test data for authenticator signature verification
+	TestKeys := []string{
+		"6cf5103c60c939a5f38e383b52239c5296c968579eec1c68a47d70fbf1d19159",
+		"0dd4d1506e18a5712080708c338eb51ecf2afdceae01e8162e890b126ac190fe",
+		"49006a359803f0602a7ec521df88bf5527579da79112bb71f285dd3e7d438033",
+	}
+	s.EncodingConfig = app.MakeEncodingConfig()
+	txConfig := s.EncodingConfig.TxConfig
 	signModeHandler := txConfig.SignModeHandler()
-	osmosisApp := app.Setup(false)
-	ak := osmosisApp.AccountKeeper
-	ctx := osmosisApp.NewContext(false, tmproto.Header{})
-
-	bz, _ := hex.DecodeString(TestKey)
-	priv1 := &secp256k1.PrivKey{Key: bz}
-	accAddress := sdk.AccAddress(priv1.PubKey().Address())
-	account1 := authtypes.NewBaseAccount(accAddress, priv1.PubKey(), 0, 0)
-	ak.SetAccount(ctx, account1)
-
-	// decode the test private key
-	bz, _ = hex.DecodeString(TestKey)
-	priv2 := &secp256k1.PrivKey{Key: bz}
-	accAddress2 := sdk.AccAddress(priv2.PubKey().Address())
-	account2 := authtypes.NewBaseAccount(accAddress2, priv2.PubKey(), 0, 0)
-	ak.SetAccount(ctx, account2)
+
+	s.OsmosisApp = app.Setup(false)
+
+	ak := s.OsmosisApp.AccountKeeper
+	s.Ctx = s.OsmosisApp.NewContext(false, tmproto.Header{})
+	s.Ctx = s.Ctx.WithGasMeter(sdk.NewGasMeter(1_000_000))
+
+	// Set up test accounts
+	for _, key := range TestKeys {
+		bz, _ := hex.DecodeString(key)
+		priv := &secp256k1.PrivKey{Key: bz}
+
+		// add the test private keys to array for later use
+		s.TestPrivKeys = append(s.TestPrivKeys, priv)
+
+		accAddress := sdk.AccAddress(priv.PubKey().Address())
+		account := authtypes.NewBaseAccount(accAddress, priv.PubKey(), 0, 0)
+		ak.SetAccount(s.Ctx, account)
+
+		// add the test accounts to array for later use
+		s.TestAccAddress = append(s.TestAccAddress, accAddress)
+	}
 
 	// Create a new Secp256k1SignatureAuthenticator for testing
-	authenticator := authenticator.NewSignatureVerificationAuthenticator(ak, signModeHandler)
-	coins := sdk.Coins{sdk.NewInt64Coin("osmo", 2500)}
+	s.SigVerificationAuthenticator = authenticator.NewSignatureVerificationAuthenticator(
+		ak,
+		signModeHandler,
+	)
+}
+
+type SignatureVerificationAuthenticatorTestData struct {
+	Msgs                               []sdk.Msg
+	AccNums                            []uint64
+	AccSeqs                            []uint64
+	Signers                            []cryptotypes.PrivKey
+	Signatures                         []cryptotypes.PrivKey
+	NumberOfExpectedSigners            int
+	NumberOfExpectedSignatures         int
+	ShouldSucceedGettingData           bool
+	ShouldSucceedSignatureVerification bool
+}
+
+type SignatureVerificationAuthenticatorTest struct {
+	Description string
+	TestData    SignatureVerificationAuthenticatorTestData
+}
+
+// TestSignatureAuthenticator test a non-smart account signature verification
+func (s *SigVerifyAuthenticationSuite) TestSignatureAuthenticator() {
```

Comment:
this is great! We can later add some other success cases with signatures in different orders and repeated signers to make sure it still works as we expect

Reply:
:muscle: 

Still want to benchmark and also test multisig!